src/UiPath.Workflow.Runtime/ParallelTrackingExtensions.cs

```diff
@@ -9,23 +9,48 @@
 /// </summary>
 public static class ParallelTrackingExtensions
 {
-    public const string BranchIdPropertyName = "__ParallelBranchId";
+    private const string BranchIdPropertyName = "__ParallelBranchId";
 
     public static ActivityInstance MarkNewParallelBranch(this ActivityInstance instance)
     {
-        var properties = new ExecutionProperties(null, instance, instance.PropertyManager);
-        var parentId = properties.Find(BranchIdPropertyName) as string;
-        properties.Add(BranchIdPropertyName, $"{parentId}.{Guid.NewGuid():N}".Trim('.'), true, false);
-
+        var parentId = instance.GetParallelBranchId();

```

Comment:
Do not modify MarkNewParallelBranch and create another method for your particular usage. SetParallelBranchId which is called now from can have side effects which are unwanted for the existing Activities.

Reply:
?